libs/sdk-common/src/invoice.rs

```diff
@@ -101,6 +101,50 @@ fn format_short_channel_id(id: u64) -> String {
     format!("{block_num}x{tx_num}x{tx_out}")
 }
 
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub enum Amount {
+    Bitcoin {
+        amount_msat: u64,
+    },
+    Currency {
+        // Reference to [FiatCurrency.id]
+        iso4217_code: String,
+        fractional_amount: u64,
```

Comment:
Would this be like Euro cents? and you can use the FiatCurrency info to calculate the decimal equivalent